googlesource/acceptance_test.go

```diff
@@ -0,0 +1,273 @@
+type AcceptanceTestDriver struct {
+	Config AcceptanceSourceTestDriverConfig
+}
+
+// AcceptanceSourceTestDriverConfig contains the configuration for
+// AcceptanceTestDriver.
+type AcceptanceSourceTestDriverConfig struct {
+	// Connector is the connector to be tested.
+	Connector sdk.Connector
+
+	// SourceConfig config for source
+	SourceConfig map[string]string
+
+	// BeforeTest is executed before each acceptance test.
+	BeforeTest func(t *testing.T)
+	// AfterTest is executed after each acceptance test.
+	AfterTest func(t *testing.T)
+
+	// GoleakOptions will be applied to goleak.VerifyNone. Can be used to
+	// suppress false positive goroutine leaks.
+	GoleakOptions []goleak.Option
+
+	// Skip is a slice of regular expressions used to identify tests that should
+	// be skipped. The full test name will be matched against all regular
+	// expressions and the test will be skipped if a match is found.
+	Skip []string
+}
+
+func (d AcceptanceTestDriver) DestinationConfig(*testing.T) map[string]string {
+	return map[string]string{}
+}
+func (d AcceptanceTestDriver) Connector() sdk.Connector {
+	return d.Config.Connector
+}
+
+func (d AcceptanceTestDriver) SourceConfig(*testing.T) map[string]string {
+	return d.Config.SourceConfig
+}
+
+func (d AcceptanceTestDriver) BeforeTest(t *testing.T) {
+	// before test check if the test should be skipped
+	d.Skip(t)
+
+	if d.Config.BeforeTest != nil {
+		d.Config.BeforeTest(t)
+	}
+}
+
+func (d AcceptanceTestDriver) AfterTest(t *testing.T) {
+	if d.Config.AfterTest != nil {
+		d.Config.AfterTest(t)
+	}
+}
+
+func (d AcceptanceTestDriver) Skip(t *testing.T) {
+	var skipRegexs []*regexp.Regexp
+	for _, skipRegex := range d.Config.Skip {
+		r := regexp.MustCompile(skipRegex)
+		skipRegexs = append(skipRegexs, r)
+	}
+
+	for _, skipRegex := range skipRegexs {
+		if skipRegex.MatchString(t.Name()) {
+			t.Skip(fmt.Sprintf("caller requested to skip tests that match the regex %q", skipRegex.String()))
+		}
+	}
+}
+
+func (d AcceptanceTestDriver) GoleakOptions(_ *testing.T) []goleak.Option {
+	return d.Config.GoleakOptions
+}
+
+// WriteToSource writes data for source to pull data from
+func (d AcceptanceTestDriver) WriteToSource(t *testing.T, records []sdk.Record) []sdk.Record {
+	var err error
+	is := is.New(t)
+	config := d.SourceConfig(t)
+	records, err = writeToSource(t, config, records)
+	is.NoErr(err)
+
+	return records
+}
+
+func writeToSource(t *testing.T, config map[string]string, records []sdk.Record) (result []sdk.Record, err error) {
+	result, err = dataSetupWithRecord(t, config, records)
+	return result, err
+}
+func (d AcceptanceTestDriver) ReadFromDestination(*testing.T, []sdk.Record) []sdk.Record {
+	return []sdk.Record{}
+}
+
+func (d AcceptanceTestDriver) GenerateRecord(_ *testing.T) sdk.Record {
+	// we don't create record over here. Because we need to code here and then decode again
+	return sdk.Record{}
+}
+
+func createDatasetForAcceptance(t *testing.T, tableID string) (err error) {
```

Comment:
What's the difference between this function, and the one in source_integration_test?